src/common/telemetry-data-factory.ts

```diff
@@ -523,4 +538,43 @@ export class TelemetryDataFactory {
             enabled,
         };
     }
+
+    public forCompletedTestStep(
+        assessment: Assessment,
+        storeData: AssessmentStoreData,
+        step: Requirement,
+    ): RequirementStatusTelemetryData {
+        const assessmentData = assessment
+            .getVisualizationConfiguration()
+            .getAssessmentData(storeData);
+        const numInstances = this.getNumInstances(step, assessmentData);
+        const newStatus = storeData.assessments[assessment.key].testStepStatus;
+        return this.forRequirementStatus(
+            assessment.visualizationType,
+            step.key,
+            newStatus[step.key].stepFinalResult === ManualTestStatus.PASS,
+            numInstances,
+            step.getCompletedStepDetailsForTelemetry
```

Comment:
I think it would be better to add this default logic to AssessmentBuilder rather than here as I don't know if it's helpful for the telemetry data factory to know so much about assessment data.

Reply:
I think if we also did do some sort of null pattern in the AssessmentBuilder we can likely avoid that if block on line 214 as well.